### PR TITLE
refactor: tests to rtl

### PR DESCRIPTION
**Pull request by nataliepina**
Refactored the TextInputWithIcon and Modal tests

# Description

These are the final refactors from enzyme to RTL. After this and the other open PRs are merged we can remove enzyme. 

## Checklist

- [ ] If any new components were added, there are exported from `packages/index.ts`
- [ ] This PR is associated with a JIRA and mentions in the commit message footer ("Closes …")
- [ ] This PR contains breaking changes and states in the commit message body ("BREAKING CHANGE: …")
- [ ] I have reviewed the changes and provided detail to the sections above

### CODE REVIEW

File: packages/modal/tests/Modal.test.tsx [hunk 1]
```diff
@@ -13,44 +14,44 @@ describe("Modal", () => {
...
+      const { getByRole } = render(
         <ModalBase isOpen={true} onClose={action}>
           <div tabIndex={0}>I am modal content</div>
         </ModalBase>
       );
+      const closeButton = getByRole("button");
 
       expect(action).not.toHaveBeenCalled();
-      component.find('[role="button"]').first().simulate("click");
+      fireEvent.click(closeButton);
```
Comment (vacas5) [inline packages/modal/tests/Modal.test.tsx:47]: did you want to try `userEvent.click` here?

**Review by vacas5 — APPROVED** [review]
think this is totally fine. If you wanted to try userEvent instead of fireEvent you could in some places.

File: packages/modal/tests/Modal.test.tsx (hunk 1 above)
Comment (nataliepina) [reply packages/modal/tests/Modal.test.tsx:47]: Not really. I did mix and match in this test, I wanted userEvent to use a key press. I've generally been using fireEvent.click() elsewhere.

Comment (nataliepina) [reply packages/modal/tests/Modal.test.tsx:47]: Unless you think it's worth doing for testing purposes? Happy to update it.

**Pull request merged**